### PR TITLE
Datetime filter update

### PR DESCRIPTION
**Pull request by ShiranAvidov**
Change datetime filter parameters and adding parameter (DatetimeFormat)
Add text datetime filter
Change logzio_shipper to use threads and queues
Fix slowness when using datetime filter
Add logs
Fix tests (using requests-mock)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,17 +43,19 @@ You'll be taken to Azure Custom deployment page. Fill in all the parameters and
...
+| Datetime Finder | If file is csv/json: write the json path of the datetime field inside each log. CSV json path will always be the name of the datetime field. Json json path can be the name of the datetime field if it's in the root, or a path contains fields separated by '.' (for example: metadata.datetime, metadata\[:1].datetime). If file is text: write a regex that will get the datetime from each log. If log has many occurrences of datetime, make sure the regex will give the right one (for example: '(?:.\*?\[0-9]){2}.*?(\[0-9])' will give the third digit). If this value cannot be found inside a log, the log will be shipped to Logz.io. Leave empty if you are not using DatetimeFilter. | Required if using DatetimeFilter | NO_DATETIME_FINDER |
+| Datetime Format | The datetime format of DatetimeFilter and datetime field in each log (for example: %Y/%m/%dT%H:%M:%S%z is for 2021/11/01T10:10:10+0000 datetime). If the format is wrong, the log will be shipped to Logz.io. Leave empty if you are not using DatetimeFilter. | Required if using DatetimeFilter | NO_DATETIME_FORMAT |
```
Comment (Doron-Bargo) [inline README.md:58]: Can you rearrange it so all the required parameters will be at the beginning and the non-required will be at the end?

File: README.md [hunk 2]
```diff
@@ -43,17 +43,19 @@ You'll be taken to Azure Custom deployment page. Fill in all the parameters and
...
+| Container Name | The name of the container inside the storage account | Required | - |
+| Logs Path | The path from where blob files will trigger the Logz.io function (including subdirectories in that path). Leave empty if you want that every blob file in the container will trigger the Logz.io function. | Not Required | \<\<ContainerLogsPath\>\>/{name} |
+| Format | The format of the log files. | Required | - |
+| Logzio URL | The Logz.io listener URL fot your region. (For more details, see the regions page: https://docs.logz.io/user-guide/accounts/account-region.html) | Required | - |
+| Logzio Token | Your Logz.io logs token. (Can be retrieved from the Manage Token page.) | Required | - |
+| Multiline Regex | The regex that matches the multiline logs in text blob files. Leave empty if you do not use multiline logs in your text blob files. | Not Required | NO_REGEX |
+| Datetime Filter | Every log with datetime greater or equal to this datetime will be shipped to Logz.io (for it to take effect DatetimeFinder and DatetimeFormat must not be empty). Leave empty if you want all logs to be shipped to Logz.io. | Not Required | NO_DATETIME_FILTER |
```
Comment (Doron-Bargo) [inline README.md]: Need example what is the Datetime structure

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
lgtm
just make the Readme more clear

File: README.md (hunk 1 above)
Comment (ShiranAvidov) [reply README.md:58]: done

File: README.md (hunk 2 above)
Comment (ShiranAvidov) [reply README.md]: done

**Review by Doron-Bargo — APPROVED** [review]
Great work

**Pull request merged**